spec/services/create_next_edition_service_spec.rb

```diff
@@ -0,0 +1,118 @@
+
+    it "appends the change note details when there is a change note and a major change" do
+      current_edition = create(:edition,
+                               :published,
+                               number: 2,
+                               published_at: Date.yesterday.noon,
+                               change_note: "note",
+                               update_type: "major")
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_history.first).to match a_hash_including(
+        "note" => "note",
+        "public_timestamp" => Date.yesterday.noon.rfc3339,
+      )
+    end
+
+    it "maintains the change history when the current edition is not a major change" do
+      current_edition = create(:edition,
+                               :published,
+                               number: 2,
+                               published_at: Date.yesterday.noon,
+                               change_note: "note",
+                               update_type: "minor")
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_history).to eq([])
+    end
+
+    it "maintains the change history when current edition lacks a change note" do
+      current_edition = create(:edition,
+                               :published,
+                               number: 2,
+                               published_at: Date.yesterday.noon,
+                               change_note: "")
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_history).to eq([])
+    end
+
+    context "when the current edition is the first edition" do
+      it "doesn't append the change note details" do
+        current_edition = create(:edition,
+                                 :published,
+                                 published_at: Date.yesterday.noon,
+                                 change_note: "note")
+
+        next_edition = described_class.call(current_edition: current_edition,
+                                            user: user)
+
+        expect(next_edition.change_history).to eq([])
+      end
+    end
+
+    context "when given a discarded edition" do
+      it "updates the state of the discarded edition" do
+        live_edition = create(:edition, :published)
+        discarded_edition = create(:edition,
+                                   state: "discarded",
+                                   current: false,
+                                   document: live_edition.document)
+
+        next_edition = described_class.call(current_edition: live_edition,
+                                            user: user, discarded_edition: discarded_edition)
+
+        expect(next_edition.state).to eq "draft"
```

Comment:
You could do `expect(next_edition).to be_draft` here to tap into some rspec wizardry

Reply:
@1pretz1 Expelliarmus - I'm convinced that RSpec has more wizardry than Hogwarts!  Initially, I tried `expect(next_edition.state).to be_draft` but that caused a failure with `expected "draft" to respond to 'draft?'`.  Then I noticed the typo (`.state`), so then `expect(next_edition).to be_draft` and it works!  You're gonna have to explain this one to me.

Reply:
I believe I found why this works and it's due to statuses state enum being delegated to the edition -> https://github.com/alphagov/content-publisher/blob/master/app/models/edition.rb#L56. Not as wizardly as I first though...

Reply:
Any object that has a question mark method can be used with a be_ expectation. For example if you had object x that responded to :test? you could do expect(x).to be_test.

It's pretty magical - it's defined here: https://github.com/rspec/rspec-expectations/blob/master/lib/rspec/matchers/built_in/be.rb#L181 (not that I understand much of that). Also looks like it checks plural too so could do expect(x).to be_tests